content/en/blog/_posts/2024-09-25-swap-ga-graduation.md

```diff
@@ -0,0 +1,475 @@
+configure specific cgroup v2 parameters (such as `memory.swap.max`) in a manner that will
+enable the desired swap configuration for a container. For runtimes that use control groups,
+the container runtime is then responsible for writing these settings to the container-level cgroup.
+
+## How can I monitor swap?
+
+### Node-level metric statistics
+
+Kubelet now collects
+[node-level metric statistics](/docs/reference/instrumentation/node-metrics/),
+which can be accessed at the `/metrics/resource` (which is used mainly by monitoring
+tools like Prometheus) and `/stats/summary` (which is used mainly by Autoscalers) kubelet HTTP endpoints.
+This allows clients who can directly interrogate the kubelet to
+monitor swap usage and remaining swap memory when using `LimitedSwap`.
+Additionally, a `machine_swap_bytes` metric has been added to cadvisor to show
+the total physical swap capacity of the machine.
+
+### NFD (Node Feature Discovery)
+
+[Node Feature Discovery](https://github.com/kubernetes-sigs/node-feature-discovery)
+is a Kubernetes addon for detecting hardware features and configuration.
+It can be utilized to discover which nodes are provisioned with swap.
+
+As an example, to figure out which nodes are provisioned with swap,
+use the following command:
+```shell
+kubectl get nodes -o jsonpath='{range .items[?(@.metadata.labels.feature\.node\.kubernetes\.io/memory-swap)]}{.metadata.name}{"\t"}{.metadata.labels.feature\.node\.kubernetes\.io/memory-swap}{"\n"}{end}'
+```
+
+This will result in an output similar to:
+```
+k8s-worker1: true
+k8s-worker2: true
+k8s-worker3: false
+```
+
+In this example, swap is provisioned on nodes `k8s-worker1` and `k8s-worker2`, but not on `k8s-worker3`.
+
+## Caveats
+
+Having swap available on a system reduces predictability.
+While swap can enhance performance by making more RAM available, swapping data
+back to memory is a heavy operation, sometimes slower by many orders of magnitude,
+which can cause unexpected performance regressions.
+Furthermore, swap changes a system's behaviour under memory pressure.
+Since enabling swap permits greater memory usage for workloads in Kubernetes that
+cannot be predictably accounted for, it also increases the risk of noisy neighbours
+and unexpected packing configurations, as the scheduler currently does not account for swap memory usage.
+
+The performance of a node with swap memory enabled depends on the underlying physical storage.
+When swap memory is in use, performance will be significantly worse in an I/O
+operations per second (IOPS) constrained environment, such as a cloud VM with
+I/O throttling, when compared to faster storage mediums like solid-state drives
+or NVMe.
+As swap might cause IO pressure, it is recommended to give a higher IO latency
+priority to system critical daemons. See the relevant section in the
+[recommended practices](#good-practice-for-using-swap-in-a-kubernetes-cluster) section below.
```

Comment:
Also mention that there is a big risk of a noisy neighbour when other Pods which use their RAM more often may force other processes into swapping